### PR TITLE
ci improvements

### PR DESCRIPTION
**Pull request by Fishrock123**
- run on "main"
- really check everything
- move to non-deprecated --workspace from --all
- reduce unecessary actions-rs usage
- really clippy everything
- docs without deps

+ run fmt-clippy-docs on beta

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -32,39 +33,39 @@ jobs:
...
-    - uses: actions-rs/toolchain@v1
-      with:
-        toolchain: nightly
-        components: rustfmt, clippy
-        override: true
+
+   - name: beta setup
+      run: |
+        rustup toolchain install beta
+        rustup default beta
```
Comment (yoshuawuyts) [inline .github/workflows/ci.yaml:59]: Can you say more why using `beta` makes sense here?

Comment (yoshuawuyts) [reply .github/workflows/ci.yaml:59]: Oh I guess it's just to get us off nightly?

**Review by yoshuawuyts — APPROVED** [review]
No objections; just a question

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (Fishrock123) [reply .github/workflows/ci.yaml:59]: Yeah, nightly is currently missing `rustfmt`...

**Comment by Fishrock123** [comment]
Hmm, right we'd need to actually install clippy and rustfmt for beta. How about we just use stable? it's preinstalled on the github actions images and so it would be much faster. But maybe we want to be ahead of the game with beta? Not 100% sure.

@yoshuawuyts What do you think?

**Comment by Fishrock123** [comment]
Ok, this works now, and has less complexity to just run fmt-clippy-docs on stable, as discussed in zulip.

Merging this and the equivalent PRs on the other repos.

**Pull request merged**